jerry-core/ecma/builtin-objects/ecma-builtin-json.c

```diff
@@ -898,11 +898,18 @@ ecma_builtin_json_quote (ecma_stringbuilder_t *builder_p, /**< builder for the r
 #if ENABLED (JERRY_ESNEXT)
     if (lit_is_code_point_utf16_high_surrogate (c))
     {
-      const ecma_char_t next_ch = lit_cesu8_peek_next (str_p);
-      if (lit_is_code_point_utf16_low_surrogate (next_ch))
+      if (str_p < str_end_p)
```

Comment:
Cannot we combine this check with the `lit_is_code_point_utf16_high_surrogate (c)` above?

Reply:
I think we can't because it would break an other corner case.

Reply:
Ohh, I see now.